api/src/Serializer/Normalizer/RelatedCollectionLinkNormalizer.php

```diff
@@ -111,12 +114,17 @@ public function normalize($data, $format = null, array $context = []): null|arra
                 continue;
             }
 
-            try {
-                $normalized_data['_links'][$rel] = ['href' => $this->getRelatedCollectionHref($data, $rel, $context)];
-            } catch (UnsupportedRelationException $e) {
-                // The relation is not supported, or there is no matching filter defined on the related entity
+            // If relation is a public property, this property can be checked to be a non-null value
+            $values = get_object_vars($data);
+            if (array_key_exists($rel, $values) && null == $values[$rel]) {
+                // target-value is NULL
                 continue;
             }
+
+            list($ok, $href) = $this->getRelatedCollectionHref($data, $rel, $context);
+            if ($ok) {
```

Comment:
a go programmer would expect the error handling first.
here it looks like you forgot it.

maybe:
if (!$ok) {
 continue;
}
$normalized_data['_links'][$rel] = ['href' => $href];

i know, its more verbose, but at least explicit like before.

Reply:
would you prefer the following:

```php
  $ok = $this->getRelatedCollectionHref($data, $rel, $context, out $href);
  if (!$ok) {
    continue;
  }
  $normalized_data['_links'][$rel] = ['href' => $href];
```